### PR TITLE
Add Feature Flag for Embedded KYC

### LINKED ISSUE #9252: Add Feature Flag for Embedded Component onboarding
### Description

After disucssing with @anu-rock, we felt that it would be sensible to implement a feature flag for Embedded Components in the next release of WooPayments, allowing us time to test and verify the behaviour before rolling out.

### Acceptance criteria

* Implement a feature flag in the client for Embedded onboarding.
* Keep the behaviour for onboarding at the default unless the flag is set.
* Remove the flag ahead of the 8.2 release (needs a new issue).

### PR DESCRIPTION
**Pull request by oaratovskyi**
Fixes https://github.com/Automattic/woocommerce-payments/issues/9252 

#### Changes proposed in this Pull Request

* Add a feature flag for the embedded KYC feature
* Use it in the code

#### Testing instructions

* Check that the change makes sense and tests pass.
* Go to the connect page, then complete MOX and you should be redirected to Stripe
* Reset account
* Use dev tools version from https://github.com/Automattic/woocommerce-payments-dev-tools/pull/123 and enable the feature flag
* Change client version to 8.2.0
* Go to the connect page, then complete MOX and you should be redirected to the embedded KYC
* Leave the embedded KYC and check that continue onboarding from connect page loads embedded KYC form too.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -70,19 +70,25 @@ addFilter(
...
-				__( 'Continue onboarding', 'woocommerce-payments' ),
-			],
-			navArgs: {
-				id: 'wc-payments-continue-onboarding',
-			},
-			capability: 'manage_woocommerce',
-		} );
+		// Currently under feature flag.
+		if (
+			wcpaySettings &&
```
Comment (dmallory42) [inline client/index.js:75]:
```suggestion
if ( wcpaySettings?.featureFlags?.isEmbeddedKycEnabled ) {
```

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9394` or branch name `dev/9252-embedded-kyc-add-feature-flag` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/9252-embedded-kyc-add-feature-flag) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: cedfcb98bb3e001bb6c5577376e41238e92507ea
- Build time: 2024-09-06 14:32:49 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/index.js (hunk 1 above)
Comment (oaratovskyi) [reply client/index.js:75]: thanks! changed in https://github.com/Automattic/woocommerce-payments/pull/9394/commits/db1f5a20f3bf28abe46d668897e270610d56251c

**Review by dmallory42 — APPROVED** [review]
Tested and works great. Nice job! :shipit:

**Pull request merged**